src/features/hotspots/root/HotspotsList.tsx

```diff
@@ -46,9 +51,27 @@ const HotspotsList = ({
     (state: RootState) => state.hotspots.rewards || {},
   )
   const order = useSelector((state: RootState) => state.hotspots.order)
+  const blockHeight = useSelector(
+    (state: RootState) => state.heliumData.blockHeight,
+  )
+  const syncStatuses = useSelector(
+    (state: RootState) => state.hotspots.syncStatuses,
+  )
 
   const { t } = useTranslation()
 
+  useEffect(() => {
+    orderedHotspots.forEach(({ address, status }) => {
```

Comment:
this seems like it could do a lot of fetching. one thing I talked about with @madninja was that this solution won't be great for lists of hotspots, only for a single hotspot details. because it introduces an n+1 query situation. I see that we show sync status for your hotspot list, and I'm just a little worried about wallets with 100, 300 etc hotspots in them. I know there's a better solution coming that would just deal with block heights so we wouldn't have to do the intermediary timestamp --> block height lookup. Do you think we should wait for that in the app and just go with the 1500 change for now? or is the fetch performance not too bad?

Reply:
Yeah I was thinking the same thing and chatted with Matt, since its temporary I figured it's ok for now but we may want to handle failed states so that it isn't just stuck loading or showing an incorrect sync state when things don't load properly.

Reply:
We could even maybe fallback to the gossiped height?

Reply:
I found a hotspot `strong-lipstick-rooster` that has 40 witnesses, the sync statuses load quickly. I chose to not show a loading indicator while they are being fetched. Status will just be blank until it is known.

I could see not loading the sync status of witnesses though. @cokes518 thoughts?

Reply:
I think falling back to the gossiped height might cause even more confusion 🤷

Reply:
That’s true haha. Maybe it’s fine for now. I think maybe not showing witness status might be a good idea to save all the api calls though. Some have hundreds of witnesses

Reply:
fine with me.

Reply:
ok. I'll remove witness status fetching...